### PR TITLE
Split code generated with Kotlin target into multiple files.

### PR DESCRIPTION
**Pull request by dlurton**
Issue #61.  This includes breaking changes to the API used by Gradle clients and to the CLI's command-line options.

PIG now generates one <domain>.generated.kt file per domain and cross-domain VisitorTransform, instead of dumping everything into a single file.

In the PartiQL project recently we've been working with multiple permuted ASTs and the generated code file was reaching over 80k lines.  IntelliJ's code completion and assist features ceased being able to read symbols defined in that file.  These problems went away when the PIG-generated file was split it into multiple files. (Hence, this PR.)

Also:

* Adds command-line option: -d/--output-directory, which is required when using the Kotlin target.
* Renames command-line option --output to --output-file, to avoid confusion with --output-directory.
* Refactors freemarker templates to allow code one type domain to be generated per file and to allow one file per cross-domain visitor transform.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pig-tests/src/org/partiql/pig/tests/generated/DomainAToDomainB.generated.kt [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+*/
+@file:Suppress("unused", "MemberVisibilityCanBePrivate", "FunctionName",
+"CanBePrimaryConstructorProperty", "UNNECESSARY_SAFE_CALL",
+"USELESS_ELVIS", "RemoveRedundantQualifierName", "LocalVariableName")
+
+package org.partiql.pig.tests.generated
+
+import com.amazon.ionelement.api.*
+import org.partiql.pig.runtime.*
+abstract class DomainAToDomainBVisitorTransform : DomainVisitorTransformBase() {
```
Comment (dlurton) [inline pig-tests/src/org/partiql/pig/tests/generated/DomainAToDomainB.generated.kt:15]: Could use a newline or two above this.

File: pig/src/org/partiql/pig/cmdline/Command.kt [hunk 2]
```diff
@@ -20,5 +20,5 @@ import java.io.File
 sealed class Command {
     object ShowHelp : Command()
     data class InvalidCommandLineArguments(val message: String) : Command()
-    data class Generate(val typeUniverseFile: File, val outputFile: File, val target: TargetLanguage) : Command()
+    data class Generate(val typeUniverseFile: File, val target: TargetLanguage) : Command()
```
Comment (alancai98) [inline pig/src/org/partiql/pig/cmdline/Command.kt:23]: This change along with the change in `TargetLanguage` would appear to break clients using the existing API such as PartiQL ([build.gradle](https://github.com/partiql/partiql-lang-kotlin/blob/c6e3fc651b/lang/build.gradle#L70-L73) uses Kotlin as the target language with the "outputFile" argument).

1. Would this change result in a new major version for PIG?
2. Should we add some backwards compatibility option (e.g. allow client code to use the old API to specify one target file)?

Comment (dlurton) [reply pig/src/org/partiql/pig/cmdline/Command.kt:23]: 1. Without a backward compatibility option, yes.
2. I would say that we don't have a large enough number of customers to justify the additional effort in providing a backward compatibility option.

Comment (alancai98) [reply pig/src/org/partiql/pig/cmdline/Command.kt:23]: Ok, I think as long as it's a new major version and it's simple + clear how to migrate, a backward compatibility option isn't required.

**Comment by alancai98** [comment]
I missed this from the initial review. The example to use PIG to generate Kotlin code in the [README.md](https://github.com/partiql/partiql-ir-generator/tree/eae9ddb5ed97028bf1333938376de267d0dbf3d0#generating-code) also needs to be updated.

**Review by alancai98 — APPROVED** [review]

**Pull request merged**